### DISCUSSION TITLE
Is possible to persist nvim-tree when switching between tabs?

### DISCUSSION THREAD

**Discussion by diegoulloao** [nvim-tree/nvim-tree.lua#2606] [Ideas]
I'd to see nvim-tree persisting when switching between tabs.

I think you'd understand the idea by watching this video:

https://github.com/nvim-tree/nvim-tree.lua/assets/45423661/59181612-8e0d-46d1-99b0-bb98e55b4bf5

I tried to simulate the behavior. Not sure if is feasible to implement it as an option.
What's currently happening is that I've opened nvim-tree at the side of both buffers.

**Comment by alex-courtis** [comment]
This is planned: #2255

This is not a trivial change. As usual, [Pull Requests](https://github.com/nvim-tree/nvim-tree.lua/blob/master/CONTRIBUTING.md )are most gratefully appreciated.

**Comment by mannders00** [comment]
Not exactly persisting the same instance, but I reached an acceptable workaround for me with:

```
require('nvim-tree').setup({
	tab = {
		sync = {
			open  = true,
			close = true,
		},
	},
})
```

https://github.com/user-attachments/assets/1dbeae30-c3a8-4297-bdda-024730695ebd

Which persists the tree size/state across tabs

The main caveat is that resizing the tree doesn't automatically apply to other tabs

But if you resize with `:NvimTreeResize`, then open/close, they all reset to the same common size

Closest I'm going to get I think

**Comment by daUnknownCoder** [comment]
is this the answer @diegoulloao and @alex-courtis : https://github.com/mikew/nvim-drawer?tab=readme-ov-file#nvim-tree

**Reply by alex-courtis** [reply]
> is this the answer @diegoulloao and @alex-courtis : https://github.com/mikew/nvim-drawer?tab=readme-ov-file#nvim-tree

That's the best mechanism, one that you have full control over.

#2255 will only remember the basics like size/filters etc. between tabs.
